cds-snc/cppd-medical-report-sails
# Fix data attributes issues

# What this does

There were a few issues that cropped up due to some less-than-explicit variable names, and some accidental object reference vs clone issues.

On our "CRUD" controllers (Conditions, Medications, Treatments), we were using `data` to store the `medicalReport` and `res.locals.data` (flash data) ... depending on context, this was causing issues, and tired brain yesterday came up with some unnecessarily complex ways of solving those issues.

Another issue was present but we just hadn't noticed it yet - when assigning an object to a variable, in Javascript by default, that assignment is a reference to the original object. So there were situations where we were accidentally overwriting the existing (condition/medication/treatment) object before validation even ran.

## To test
Well, some of these issues were hard to spot before, but hopefully we've got all this stuff nailed down now. To confirm, in each of the three sections (Conditions, Medications, Treatments):

- Check that the Patient Name is showing up correctly at the top of the form
- Start creating a new item but submit incomplete
- Make sure validation fired, and the changes you made previously have been repopulated
- Edit an existing item, but submit it as incomplete
- Make sure validation fired, and the changes you made previously have been repopulated
